### ISSUE TITLE
external libc crate triggers unstable library feature warnings

### ISSUE THREAD

**Issue by brson** [rust-lang/rust#21858]
I have a cargo project that links to libc 0.1.1. I see this warning

```
src/lib.rs:4:1: 4:19 warning: use of unstable library feature 'libc'
src/lib.rs:4 extern crate libc;
             ^~~~~~~~~~~~~~~~~~
src/lib.rs:4:1: 4:19 help: add #![feature(libc)] to the crate attributes to silence this warning
src/lib.rs:4 extern crate libc;
             ^~~~~~~~~~~~~~~~~~
```

This shouldn't happen with the out-of-tree crate.

**Comment by alexcrichton** [comment]
Dupe of https://github.com/rust-lang/rust/issues/21771

**Comment by alexcrichton** [comment]
(fixed by https://github.com/rust-lang/rust/pull/21782)
